### PR TITLE
put bot command link above details fold

### PR DESCRIPTION
**Pull request by crimsonfaith91**
addressing https://github.com/kubernetes/test-infra/issues/2823

### CODE REVIEW

**Comment by crimsonfaith91** [comment]
@k8s-bot ok to test

**Comment by grodrigues3** [comment]
@k8s-bot ok to test

**Comment by grodrigues3** [comment]
/assign @cblecker 

@spiffxp @kubernetes/sig-contributor-experience-pr-reviews

**Comment by grodrigues3** [comment]
Available bot commands:
		
		/assign [@userA @userB @etc]			Assign specified people (or yourself if no one is specified).
												Target must be a kubernetes org member.
		/unassign [@userA @userB @etc]			Unassign specified people (or yourself if no one is specified).
												Target must already be assigned.
		/cc [@userA @userB @etc]				Request review from specified people (or yourself if no one is specified).
												Target must be a kubernetes org member.
		/uncc [@userA @userB @etc]				Dismiss review request for specified people (or yourself if no one is specified).
												Target must already have had a review requested.
		/area [label1 label2 ...]				Add an area/<> label(s) if it exists
		/remove-area [label1 label2 ...]		Remove an area/<> label(s) if it exists
		/kind [label1 label2 ...]				Add a kind/<> label(s) if it exists
		/remove-kind [label1 label2 ...]		Remove a kind/<> label(s) if it exists
		/priority [label1 label2 ...]			Add a priority/<> label(s) if it exists
		/remove-priority [label1 label2 ...]	Remove a priority/<> label(s) if it exists
		/sig [label1 label2 ...]				Add a sig/<> label(s) if it exists
		/remove-sig [label1 label2 ...]			Remove a sig/<> label(s) if it exists
		/lgtm									Add the lgtm label
		/lgtm cancel							Remove the lgtm label
		/approve								Approve all the files for which you are an approver
		/approve cancel							Remove your approval on a pull-request
		/close									Close the issue
		/reopen									Reopen a closed issue
		/release-note							Add the release-note label
		/release-note-none						Add the release-note-none label
		@kubernetes/sig-<some-github-team>		Add the corresponding sig label
		@k8s-bot test this						Run tests defined in test-infra/prow/config.yaml
		@k8s-bot ok to test 					Allow the PR author to @k8s-bot test this
		@k8s-bot tell me a joke					Tell a bad joke, sometimes

**Comment by grodrigues3** [comment]
The output is going to look like the above.

File: prow/plugins/help/help_test.go [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+		commentNum int
+	}{
+		{
+			name:       "non-help comment",
+			action:     "created",
+			body:       "help",
+			commentNum: 0,
+		},
+		{
+			name:       "help comment",
```
Comment (grodrigues3) [inline prow/plugins/help/help_test.go]: Try `/help` in various parts of the comment.  And test multi line functionality

File: prow/plugins/help/help_test.go [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+				State:  "open",
+			},
+		}
+
+		if err := handle(fc, logrus.WithField("plugin", pluginName), ice); err != nil {
+			t.Errorf("For case %s, didn't expect error when creating help comment: %v", tc.name, err)
+			continue
+		}
+
+		commentNum := len(fc.IssueComments[5])
```
Comment (grodrigues3) [inline prow/plugins/help/help_test.go]: Where is the 5 coming from?  Can you use a descriptive variable name?

**Comment by grodrigues3** [comment]
The output is going to look like the above.  

High-level thoughts
1.  Should we make this configurable so that you can type `/help assign` and it only displays assignment related comments.  We could use some fuzzy matching perhaps for the matching.

That may be better suited for a later discussion though.  I'd be just as happy getting this in as is.

**Comment by crimsonfaith91** [comment]
@grodrigues3 
We should make the help command configurable (e.g., `/help assign`). In this case, we can change `/help` to list available options:
```
/help assign
/help unassign
/help cc
...
```

It may be better suited for later though.

File: prow/plugins/help/help_test.go (hunk 1 above)
Comment (crimsonfaith91) [inline prow/plugins/help/help_test.go]: noted!

File: prow/plugins/help/help_test.go (hunk 2 above)
Comment (crimsonfaith91) [inline prow/plugins/help/help_test.go]: it comes from `Number` under `github.Issue`.

**Comment by cblecker** [comment]
#1 - there are bazel errors.. `./verify/update-bazel.sh` should be run
#2 - I prefer outputting all commands in one go just to reduce comment spam on a ticket. github notifications would go off for anyone assigned (including the assigned reviewers).

Thinking about it, what if instead of spamming the issue/PR with the commands, we updated [this](https://github.com/kubernetes/test-infra/blob/ba1d00153657c7888785f547b9afd6cb9fab0a1f/prow/plugins/trigger/pr.go#L76-L96) to add a link. That way new contributors and non-members would get the link to the commands right off the bat.

Most if not all k8s org members should already know about the bot commands, or at least where to look for them.

**Comment by crimsonfaith91** [comment]
For now, I set up the simplest regex for the /help command: regexp.MustCompile(`/help`)
We can elaborate on it later on when extending the command.

**Comment by crimsonfaith91** [comment]
@cblecker The bazel error should have been fixed now.
I lean towards having both `/help` and a link at `askToJoin` prompt. When the conversation becomes long, it will be easier if i do not need to always scroll up to the top of issue / PR for accessing the link.

**Comment by grodrigues3** [comment]
@cblecker 

> Thinking about it, what if instead of spamming the issue/PR with the commands, we updated this to add a link.

I'm fine with adding a link there, but I think we should also have the `/help` functionality.  askToJoin func is only called for PRs and not issues.  The assign and label adding commands are very important for issues and I want to make sure that new users can easily discover them.

If it gets too spammy we can disable or modify it.

**Comment by grodrigues3** [comment]
/lgtm

But I'll wait for @cblecker's opinion before merging

**Comment by fejta** [comment]
You need to enable it in plugins.yaml, also you need to ensure it is compiled into hook and then ./bump.sh hook

**Comment by cblecker** [comment]
Perhaps we want a similar function to askToJoin for issues too..

I'm okay with trialing this, but I'm not 100% sure this will solve the problem it's trying to. How would new users know how to use `/help` to begin with? To me, having either: a comment on the PR/issue (preferably before assigning reviewers) saying "Welcome to k8s! I'm a bot and my commands are here" would help onboard users more than having a command that outputs point in time copies of a document into the ticket itself.

On implementation.. is there a reason the text is not aligned? We could have the commands either display as preformatted/monospace (but you need the three backticks to do that), or we could have them as formatted markdown table.

**Comment by grodrigues3** [comment]
> "Welcome to k8s! I'm a bot and my commands are here" would help onboard users more than having a command that outputs point in time copies of a document into the ticket itself.

Let's do both.  I think this is a small enough and related enough change that this could be added to this PR. @crimsonfaith91

**Comment by rmmh** [comment]
You should use a table instead. It's easier to read, and will format itself nicely.

**Comment by spiffxp** [comment]
I'm with @cblecker... we're all aware the bot already posts "I understand the comments posted [here](https://github.com/kubernetes/test-infra/blob/master/commands.md)." right?  See the first bot comment in this thread

**Comment by rmmh** [comment]
Including the documentation link in the "ok to test" message (not inside the fold) might get much of the same utility, without a new plugin.

**Comment by crimsonfaith91** [comment]
I will add the link in the `ok to test` message. Will having both `/help` and the link work better?

**Comment by crimsonfaith91** [comment]
+1 for having a link above details fold for the `ok to test` message. i agree that having the help plugin is unnecessary. we can scroll up to top of the PR page quickly to get the information.

from my understanding, `ok to test` message does not appear for issues. how should we get the information to the issue side? my current idea is to create a munger to add the bot comment information, but it may be overkill in this case.

@grodrigues3 @rmmh @cblecker @spiffxp

File: prow/plugins/respond.go [hunk 3]
```diff
@@ -23,7 +23,7 @@ import (
 	"k8s.io/test-infra/prow/github"
 )
 
-const AboutThisBot = "Instructions for interacting with me using PR comments are available [here](https://github.com/kubernetes/community/blob/master/contributors/devel/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://github.com/kubernetes/test-infra/blob/master/commands.md)."
```
Comment (rmmh) [inline prow/plugins/respond.go]: You shouldn't have deleted this here. AboutThisBot is used in more plugins than just the "OK to test" comment, so removing this piece makes *other* bot responses lose the details below the fold.

This should be:

    const AboutThisBotWithoutCommands = "Instructions for interacting with me using PR comments are available [here](https://github.com/kubernetes/community/blob/master/contributors/devel/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository."
    const AboutThisBotCommands = "I understand the commands that are listed [here](https://github.com/kubernetes/test-infra/blob/master/commands.md)."
    const AboutThisBot = AboutThisBotWithoutCommands + " " + AboutThisBotCommands

Comment (crimsonfaith91) [inline prow/plugins/respond.go]: @rmmh i think i got your meaning. for other comments, the link should still be there. am i right about this?
i thought we are removing the link for _all_ comments.

Comment (crimsonfaith91) [inline prow/plugins/respond.go]: what i do now is breaking down into 2 parts in `respond.go`, and then call `plugins.AboutThisBotCommands` in `pr.go`.

**Pull request merged**